### PR TITLE
Fix `prevent-abbreviations` fixer bug

### LINKED ISSUE #330: `prevent-abbreviations` should not try to fix variable names when they're being destructured
First off, I think this is a great library with a lot of excellent rules and really appreciate all the work on it.

I was working on implementing some of the rules for my team and I ran an autofix and saw this change in my git diff:

image

In this case, it would definitely introduce a bug since there 's no 'development' variable being passed.

While I think this autofix is generally pretty safe, i think it might be good to check if the current scope includes destructuring and avoid autofixing.

I can look into working on a PR for this myself (i've been interested in learning how to work on eslint rules), just wanted to point this out and see if i can help out.

### PR DESCRIPTION
**Pull request by fisker**
fixes: #443 and fixes #330

I've located the problem, in `isShorthandPropertyIdentifier` check function, In `babel-eslint` parent.key is not reference of identifier, add a extra check to make it pass

### CODE REVIEW

File: rules/prevent-abbreviations.js [hunk 1]
```diff
@@ -366,8 +366,17 @@ const shouldFix = variable => {
...
+		identifier.parent.shorthand &&
+		(
+			identifier.parent.key === identifier ||
+			// In `babel-eslint` parent.key is not reference of identifier
+			(
+				identifier.parent.key.type === identifier.type &&
+				identifier.parent.key.name === identifier.name &&
+				identifier.parent.key.start === identifier.start &&
+				identifier.parent.key.end === identifier.end
+			)
```
Comment (fisker) [inline rules/prevent-abbreviations.js]: I don't have much experience on those `AST`s, I'm not sure if this is good

**Comment by fisker** [comment]
also there is a few other functions in this file compares `parent.key === identifier`, I'm not sure those logic requires modification

File: rules/prevent-abbreviations.js [hunk 2]
```diff
@@ -366,8 +366,17 @@ const shouldFix = variable => {
 
 const isShorthandPropertyIdentifier = identifier => {
 	return identifier.parent.type === 'Property' &&
-		identifier.parent.key === identifier &&
-		identifier.parent.shorthand;
+		identifier.parent.shorthand &&
+		(
+			identifier.parent.key === identifier ||
+			// In `babel-eslint` parent.key is not reference of identifier
```
Comment (sindresorhus) [inline rules/prevent-abbreviations.js]: Can you open an issue about this on `babel-eslint`? The point of that package is to normalize the AST.

File: rules/prevent-abbreviations.js [hunk 3]
```diff
@@ -366,8 +366,17 @@ const shouldFix = variable => {
...
+			(
+				identifier.parent.key.type === identifier.type &&
+				identifier.parent.key.name === identifier.name &&
+				identifier.parent.key.start === identifier.start &&
```
Comment (sindresorhus) [inline rules/prevent-abbreviations.js]: You should not rely on `.start` and `.end`: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/318

File: rules/prevent-abbreviations.js (hunk 2 above)
Comment (fisker) [reply rules/prevent-abbreviations.js]: you are right, we might not do this check, I've found other cases also broken, let me have a look at their repo first

Comment (sindresorhus) [reply rules/prevent-abbreviations.js]: I’m ok with working around it, but we should always also try to get it fixed upstream.

Comment (sindresorhus) [reply rules/prevent-abbreviations.js]: Add a code comment with a link to the Babel-eslint issue.

**Comment by fisker** [comment]
i just confirmed `(Property)Node.key.parent ==== Node`, so it's maybe not `babel-eslint` bug, let me dig more deeper

**Pull request merged**